Move AllowClientRenegotiation tests

richsalz opened the pull request:
Move them from test_renegotiation to renegotiation in ssl_new

richsalz commented:
Not passing :(  Any clue appreciated.

mattcaswell commented on test/ssl-tests/17-renegotiate.cnf.in:
```diff
@@ -261,6 +261,37 @@ our @tests_tls1_2 = (
+        test => {
+            "Method" => "TLS",
+            "HandshakeMode" => "RenegotiateClient",
+            "ResumptionExpected" => "No",
```
This code:

https://github.com/openssl/openssl/blob/eb1b66f00ca4e1fb6f9e815e8686768b6d81722d/test/helpers/handshake.c#L894-L919

Suggests that the answer here should be "Yes"

mattcaswell commented on test/ssl-tests/17-renegotiate.cnf.in:
```diff
@@ -261,6 +261,37 @@ our @tests_tls1_2 = (
+        name => "allow-client-renegotiation",
+        server => {
+            "MaxProtocol" => "TLSv1.2",
+            "Options" => "ClientRenegotiation",
```
You previously changed the test to turn this option on by default:

https://github.com/openssl/openssl/blob/eb1b66f00ca4e1fb6f9e815e8686768b6d81722d/test/ssl_test.c#L454-L481

richsalz replied:
>you previous changed the test to turn this option on by default.

Schizophrenia? Senility?  Aargh.

mattcaswell commented on test/ssl-tests/17-renegotiate.cnf.in:
```diff
@@ -261,6 +261,37 @@ our @tests_tls1_2 = (
+    {
+        name => "no-client-renegotiation",
+        server => {
+            "MaxProtocol" => "TLSv1.2"
```
Since `SSL_OP_ALLOW_CLIENT_RENEGOTIATION` is "on" by default for this test, you need to turn it off again, e.g.

````
            "Options" => "-ClientRenegotiation",
````

richsalz commented:
```
    # ERROR: (int) 'result->result == test_ctx->expected_result' failed @ test/ssl_test.c:36
    # [0] compared to [1]
    # INFO:  @ test/ssl_test.c:39
    # ExpectedResult mismatch: expected ServerFail, got Success.
    # ERROR:  @ test/ssl_test.c:92
    # ServerAlert mismatch: expected no renegotiation, got close notify.
    # 
    # ERROR: (int) 'result->client_resumed == test_ctx->resumption_expected' failed @ test/ssl_test.c:219
    # [1] compared to [0]
    # OPENSSL_TEST_RAND_ORDER=1623179872
    not ok 17 - iteration 17
```

I'm about to just drop this.  Any clues?

kaduk commented on test/ssl-tests/17-renegotiate.cnf.in:
```diff
@@ -261,6 +261,37 @@ our @tests_tls1_2 = (
+        },
+        client => {
+            "MaxProtocol" => "TLSv1.2",
+            "Options" => "-ClientRenegotiation"
```
Doesn't `-ClientRenegotiation` only actually do anything when set in the server's configuration?

richsalz replied:
Could you hear the sound of my head hitting the desk from way over where you were?

richsalz commented:
Rebased.  All tests pass.  Added a commit to remove the "-immediate_renegotiation" flag and support, since it is not being used because the tests is now done in the SSL `17-renegotiate.cnf` tests as @mattcaswell requested.

kaduk commented:
>  richsalz requested review from mattcaswell and kaduk 11 hours ago

FWIW, I don't expect to be able to put time into review this week

mattcaswell commented:
LGTM.

Ping for second review.

richsalz commented:
@kaduk, I didn't really expect you to do a full review as we both work for Akamai.  It was just to see that your drive-by comment was exactly on-target.

openssl-machine commented:
24 hours has passed since 'approval: done' was set, but as this PR has been updated in that time the label 'approval: ready to merge' is not being automatically set.  Please review the updates and set the label manually.

t8m reviewed (approved):
Merging.

t8m commented:
Merged to master. Thank you for the contribution.
